### PR TITLE
gts: regenerate `configure` for `-flat_namespace` audit

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

`gts` currently fails our `-flat_namespace` audit, but none of our usual
patches apply. We can fix this by regenerating `configure` instead with
our Autotools stack.

While we're here, let's add a test since this formula is missing one.

This is needed for bottling on Monterey.

### CODE REVIEW

**Comment by carlocab** [comment]
License ref: https://metadata.ftp-master.debian.org/changelogs//main/g/gts/gts_0.7.6+darcs121130-5_copyright

**Comment by SMillerDev** [comment]
We should also really deprecate this since the last update was > 10 years ago

File: Formula/gts.rb [hunk 1]
```diff
@@ -14,7 +15,13 @@ class Gts < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "8574a923a6854116b836f04ffdd0b556d71c3e63da1127a90c2e6ac573900c88"
   end
 
-  depends_on "pkg-config" => :build
+  # We regenerate configure to avoid the `-flat_namespace` flag.
```
Comment (SMillerDev) [inline Formula/gts.rb:18]: I think this was supposed to be on a different line?

Comment (carlocab) [reply Formula/gts.rb:18]: No, I did mean to put this here. I was trying to explain the Autotools dependencies.

**Comment by carlocab** [comment]
> We should also really deprecate this since the last update was > 10 years ago

Actually, I don't think this is a good idea. Deprecated formulae get skipped by `test-bot`: https://github.com/Homebrew/homebrew-test-bot/blob/6cc7d8dcce15757e7ada59d972f9a718cea3b3db/lib/tests/formulae_dependents.rb#L124-L128

This would normally be fine, but we still have a lot of formulae that depend on this one, and a `linkage` failure for `gts` means all these formulae could get broken:
```
anjuta                   evince                   gnome-latex              gtranslator              plantuml                 vala-language-server
cherrytree               fobis                    gnome-recipes            makefile2graph           qcachegrind              valabind
dnsviz                   ford                     gprof2dot                mlpack                   root                     zim
doteditor                gammaray                 graphviz                 msc-generator            simgrid
eralchemy                gedit                    gspell                   msitools                 vala
```
We've rev bumped `gts` a few times recently, so those linkage failures do happen.

File: Formula/gts.rb (hunk 1 above)
Comment (carlocab) [reply Formula/gts.rb:18]: Ah, wait, I see what you mean. `brew style --fix` moved my comment 😅

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**